zulipterminal/ui_tools/boxes.py

```diff
@@ -810,6 +812,17 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
         elif is_command_key("MARKDOWN_HELP", key):
             self.view.controller.show_markdown_help()
             return key
+        elif is_command_key("OPEN_IN_TERMINAL_EDITOR", key):
+            edit_tempfile = NamedTemporaryFile(suffix=".md")
+            with open(edit_tempfile.name, mode="w") as edit_writer:
+                edit_writer.write(self.msg_write_box.edit_text)
+            self.view.controller.loop.screen.stop()
+            subprocess.call([f"$EDITOR {edit_tempfile.name}"], shell=True)
+            with open(edit_tempfile.name, mode="r") as edit_reader:
+                self.msg_write_box.edit_text = edit_reader.read()
+            edit_tempfile.close()
+            self.view.controller.loop.screen.start()
```

Comment:
One potential concern I have here is whether there is anything that could block getting to this point and restarting the application loop.

Reply:
I don't think this is really a problem, the only case it will lock is if the external editor freeze and in that case the user would already try to kill it unlocking zulip.

Reply:
I agree this is OK for now, though if an external editor is opened (eg. gvim) then I can switch back to the application and it allows me to keep typing.

That may or may not be related to the issue with gvim not working more generally - it leaves any text I entered *while gvim was open* in the text box. Maybe since it is running in the background, the subprocess exits already?

Reply:
Exactly, in that case, the gvim command will close directly. Maybe like for sublime text and lapce there is a `--watch` and `--new` flags to use in the command.